pytorch_lightning/trainer/trainer.py

```diff
@@ -1072,13 +1079,22 @@ def _run_sanity_check(self, ref_model):
             # restore the previous stage when the sanity check if finished
             self.state.stage = stage
 
-    def __load_ckpt_weights(self, ckpt_path: Optional[str]) -> Optional[str]:
-        if ckpt_path is None:
-            return
+    def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_connected: bool) -> Optional[str]:
+        if model_provided and ckpt_path is None:
+            return  # use passed model to function without loading weights
 
         fn = self.state.fn.value
 
-        if ckpt_path == 'best':
+        if model_connected and ckpt_path is None:
+            rank_zero_warn(
+                f"`.{fn}(ckpt_path=None)` was called without a model. "
+                f"The best model of the previous `fit` call will be used. "
+                f"You can pass `ckpt_path='best'` to avoid this warning "
+                f"or `ckpt_path=trainer.model_checkpoint.last_model_path` to use the last model."
```

Comment:
```suggestion
                "The best model of the previous `fit` call will be used. "
                f"You can pass `{fn}(ckpt_path='best')` to avoid this warning "
                "or `ckpt_path=trainer.model_checkpoint.last_model_path` to use the last model."
```